ddtrace/internal/telemetry/data/metrics.py

```diff
@@ -0,0 +1,58 @@
+import time
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from ...hostname import get_hostname
+
+
+class Series:
+    """
+    Stores and sends metrics using the Datadog Metrics API: https://docs.datadoghq.com/api/latest/metrics/
+    """
+
+    GAUGE = "gauge"
+    COUNT = "count"
+    RATE = "rate"
+
+    def __init__(self, metric, metric_type=COUNT, common=False, interval=None):
+        # type: (str, str, bool, Optional[int]) -> None
+        self.metric = metric
+        self.type = metric_type
+        self.interval = interval
+        self.common = common
+        self.points = []  # type: List[Tuple[int, int]]
+        self.tags = {}  # type: Dict[str, str]
+        self.host = get_hostname()
+
+    def add_point(self, value):
+        # type: (int) -> None
+        """
+        Adds timestamped data point associated with a metric
+        """
+        timestamp = int(time.time())  # type: int
+        self.points.append((timestamp, value))
+
+    def add_tag(self, name, value):
```

Comment:
not a big deal, but we use `set_tag` on `Span`, do we want to mimic that here?

this really isn't a big deal, not sure I care too much, just a thought.